jax-stuff.md

```diff
@@ -77,51 +78,52 @@ _styles: >
 
 ## How Does Parallelism Work in JAX?
 
-JAX supports two schools of thought for multi-device programming:
+JAX supports three levels of parallelism for multi-device programming:
 
-1. **Compiler, take the wheel!** Let the compiler automatically partition arrays and decide what communication to add to facilitate a given program. This lets you write a program on a single device and automatically run it on hundreds without changing anything. 
-2. **Just let me write what I mean, damnit!** While compilers are nice, they sometimes do the wrong thing and add communication you don't intend. Sometimes we want to be extremely explicit about what we're doing.
+1. **Compiler-based automatic parallelization** You program as if using a single global machine, and the compiler chooses how to shard data and partition computation (with some user-provided constraints).
+2. **Explicit sharding and automatic partitioning** You still have a global view but data shardings are
+explicit in JAX types, inspectable using `jax.typeof`.
```

Comment:
nit: maybe something like "Automatic parallelism is great, but sometimes the compiler does something crazy. Explicit sharding lets you write single-device code like usual, but have JAX itself handle sharding propagation (not the compiler). This means JAX can ask you for clarification when it's unclear what you want."